### ISSUE TITLE
Add `card` parameter to `UpdatePaymentMethod`

### ISSUE THREAD

**Issue by mwilliammyers** [arlyon/async-stripe#85]
[`stripe::UpdatePaymentMethod`](https://docs.rs/async-stripe/0.13.0-rc3/stripe/struct.UpdatePaymentMethod.html) (and the master branch) seems to be missing the [`card`](https://stripe.com/docs/api/payment_methods/update#update_payment_method-card) parameter, to e.g. update an already saved card's expiration without needing to create a new card (with the full number).

Can it be added? (I can try to open a PR). Or am I missing it somewhere else?

**Comment by Parth** [comment]
👋 `card` is also missing for `CreatePaymentMethod`, we'll probably need to roll our own implementation while this is missing.

**Comment by arlyon** [comment]
Essentially the crux of the issue is that stripe doesn't declare cards as a reference in the OpenAPI schema and instead defines them individually in a number of places (at least 7, maybe more).

So we are left with a dilemma because we can't generate a `Card` type since they may differ in specific ways (and stripe provides no guarantees that new 'card' types follow the same format). An example of these differing Card impls:

```json
"card": {
  "anyOf": [
    {
      "properties": {
        "address_city": {
          "maxLength": 5000,
          "type": "string"
        },
        "address_country": {
          "maxLength": 5000,
          "type": "string"
        },
        "address_line1": {
          "maxLength": 5000,
          "type": "string"
        },
        "address_line2": {
          "maxLength": 5000,
          "type": "string"
        },
        "address_state": {
          "maxLength": 5000,
          "type": "string"
        },
        "address_zip": {
          "maxLength": 5000,
          "type": "string"
        },
        "cvc": {
          "maxLength": 5000,
          "type": "string"
        },
        "exp_month": {
          "type": "integer"
        },
        "exp_year": {
          "type": "integer"
        },
        "metadata": {
          "additionalProperties": {
            "type": "string"
          },
          "type": "object"
        },
        "name": {
          "maxLength": 5000,
          "type": "string"
        },
        "number": {
          "maxLength": 5000,
          "type": "string"
        },
        "object": {
          "enum": [
            "card"
          ],
          "maxLength": 5000,
          "type": "string"
        }
      },
      "required": [
        "exp_month",
        "exp_year",
        "number"
      ],
      "title": "customer_payment_source_card",
      "type": "object"
    },
    {
      "maxLength": 5000,
      "type": "string"
    }
  ],
  "description": "A token, like the ones returned by [Stripe.js](https://stripe.com/docs/js).",
  "x-stripeBypassValidation": true
},
```

```json
"card": {
  "anyOf": [
    {
      "properties": {
        "cvc": {
          "maxLength": 5000,
          "type": "string"
        },
        "exp_month": {
          "type": "integer"
        },
        "exp_year": {
          "type": "integer"
        },
        "number": {
          "maxLength": 5000,
          "type": "string"
        }
      },
      "required": [
        "exp_month",
        "exp_year",
        "number"
      ],
      "title": "card_details_params",
      "type": "object"
    },
    {
      "properties": {
        "token": {
          "maxLength": 5000,
          "type": "string"
        }
      },
      "required": [
        "token"
      ],
      "title": "token_params",
      "type": "object"
    }
  ],
  "description": "If this is a `card` PaymentMethod, this hash contains the user's card details. For backwards compatibility, you can alternatively provide a Stripe token (e.g., for Apple Pay, Amex Express Checkout, or legacy Checkout) into the card hash with format `card: {token: \"tok_visa\"}`. When providing a card number, you must meet the requirements for [PCI compliance](https://stripe.com/docs/security#validating-pci-compliance). We strongly recommend using Stripe.js instead of interacting with this API directly.",
  "x-stripeBypassValidation": true
},
```

**Comment by imalsogreg** [comment]
@arlyon Would you take a PR that gives a one-off solution to the problem of creating a Payment method with a card? For example 

```

pub struct PaymentMethodCard {
  pub exp_year: u16,
  pub exp_month: u8,
  pub number: String,
  pub ccv: u16
}

impl PaymentMethod {
  ...
  pub fn create_with_card( client: &Client, card: PaymentMethodCard ) -> Response<PaymentMethod> {
   ...
  }

  pub fn update_with_card( client: &Client, id: PaymentMethodId, card: PaymentMethodCard) -> Response<PaymentMethod> {
    ...
  }
}
```

This may unblock some of the users who need specific card stuff, and could be deleted if someone more ambitious implements the fully general solution in your proposal.

**Comment by erichCompSci** [comment]
Hey, this is already in mainline, by the way.  I fixed it a while back, just didn't close the issue.
